crates/trigger-http/src/instrument.rs

```diff
@@ -24,39 +25,43 @@ macro_rules! http_span {
         )
     };
 }
+
 pub(crate) use http_span;
 
-type ResponseResult = Result<Response<HyperIncomingBody>>;
+/// TODO
+pub struct RoutedResponse(pub Response<HyperIncomingBody>, pub Option<String>);
 
-pub(crate) fn finish_instrumenting(
-    response: ResponseResult,
+/// TODO
+pub(crate) fn finalize_http_span(
+    foo: Result<RoutedResponse>,
     method: String,
-    route: Option<&str>,
-) -> ResponseResult {
+) -> Result<Response<HyperIncomingBody>> {
     let span = tracing::Span::current();
+    match foo {
+        Ok(RoutedResponse(response, route)) => {
+            // Set otel.name and http.route
+            if let Some(route) = route {
+                span.record("http.route", &route);
+                span.record("otel.name", format!("{method} {route}"));
+            } else {
+                span.record("otel.name", method);
```

Comment:
```suggestion
                span.record("otel.name", "{method} unknown");
```

would be easier to filter for in trace management software

Reply:
I agree on the suggestion, but slight nit:
```suggestion
                span.record("otel.name", format!("{method} <unknown route>");
```

Reply:
The value as-is matches the [OTel semantic conventions](https://opentelemetry.io/docs/specs/semconv/http/http-spans/#name)

> If there is no (low-cardinality) `http.route` available, HTTP server span names SHOULD be `{method}`.

Reply:
^ Yep, I'm just trying to follow the OTEL semantic conventions here.